src/builddedicatedserver

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+set -euo pipefail
+
+script=$(readlink -f -- "$0")
+pushd "$(dirname -- "$script")" > /dev/null
+
+source sdk_container
+run_in_sniper "$@"
+
+if [ $# -eq 0 ]; then
+    export VPC_NINJA_BUILD_MODE="release"     
+else
+    if [[ "$1" == "debug" ]]; then
+        export VPC_NINJA_BUILD_MODE="debug"
+    elif [[ "$1" == "release" ]]; then
+        export VPC_NINJA_BUILD_MODE="release"
+    else
+        echo "Usage: $0 [debug|release]"
+        exit 1
+    fi
+fi
+
+solution_out="_vpc_/ninja/sdk_server_$VPC_NINJA_BUILD_MODE"
+
+if [[ ! -e "$solution_out.ninja" ]]; then
+    devtools/bin/vpc /hl2mp /tf /linux64 /ninja /define:SOURCESDK +dedicated /mksln "$solution_out"
+
+    # Generate compile commands.
+    ninja -f "$solution_out.ninja" -t compdb > compile_commands.json
+
+    # Remove some unsupported clang commands.
+    sed -i 's/-fpredictive-commoning//g; s/-fvar-tracking-assignments//g' compile_commands.json
+    sed -i 's|/my_mod/src|.|g' compile_commands.json
+
+    # Reflect changes done in compile_commands.json for easier diagnostics
+    sed -i 's|-D_DLL_EXT=\.so|-D_DLL_EXT=_srv.so|g' compile_commands.json
+    sed -i 's|-D_EXTERNAL_DLL_EXT=\.so|-D_EXTERNAL_DLL_EXT=_srv.so|g' compile_commands.json
+    sed -i 's|server\.so|server_srv.so|g' compile_commands.json
+    sed -i 's|lvstdlib|lvstdlib_srv|g' compile_commands.json
+    sed -i 's|ltier0|ltier0_srv|g' compile_commands.json
+
+    sed -i 's|-D_DLL_EXT=\.so|-D_DLL_EXT=_srv.so|g' "$solution_out.ninja"
+    sed -i 's|-D_EXTERNAL_DLL_EXT=\.so|-D_EXTERNAL_DLL_EXT=_srv.so|g' "$solution_out.ninja"
+    sed -i 's|server\.so|server_srv.so|g' "$solution_out.ninja"
+    sed -i 's|libtier0\.so|libtier0_srv.so|g' "$solution_out.ninja"
+    sed -i 's|libvstdlib\.so|libvstdlib_srv.so|g' "$solution_out.ninja"
+    sed -i 's|lvstdlib|lvstdlib_srv|g' "$solution_out.ninja"
+    sed -i 's|ltier0|ltier0_srv|g' "$solution_out.ninja"
```

Comment:
This feels *really* hacky, is it be possible to instead use the proper `/dedicated` option for VPC?

Reply:
Darn i guess that would be easier just to copy every `.a` file in the /lib/common and /lib/public to have suffix `_srv.a`
still to run server properly, from testing now it's neccessary to have server libraries for libtier0 and libvstdlib otherwise random things start to break like ClientCommandKeyValues.